packages/data-explorer/src/store/actions.ts

```diff
@@ -68,34 +83,32 @@ export default {
     }
 
     let columns: string[]
-    let tableData
+    let request
     const isCustomCard = state.dataDisplayLayout === 'CardView' && state.tableSettings.customCardCode
-
     const rsqlQuery = getters.filterRsql
 
     commit('setTableData', [])
+
     if (isCustomCard) {
       columns = state.tableSettings.customCardAttrs
         .split(',')
         .filter(f => f !== '')
         .map(a => a.trim())
 
-      tableData = await dataRepository.getTableDataDeepReference(
-        state.tableName,
-        state.tableMeta,
-        columns,
-        rsqlQuery,
-        state.dataDisplayLimit)
+      request = await dataRepository.getTableDataDeepReference(state.tableName, state.tableMeta, columns, rsqlQuery, pagination)
     } else {
       columns = metaDataService.getAttributesfromMeta(state.tableMeta).splice(0, state.tableSettings.collapseLimit)
-      tableData = await dataRepository.getTableDataWithLabel(state.tableName, state.tableMeta, columns, rsqlQuery, state.dataDisplayLimit)
+      request = await dataRepository.getTableDataWithLabel(state.tableName, state.tableMeta, columns, rsqlQuery, pagination)
     }
+
     if (getters.filterRsql === rsqlQuery) {
       // retrieved results are still relevant
-      commit('setTableData', tableData)
+      commit('setTableData', request)
     }
+
+    return request
```

Comment:
why is the request itself exposed to the outside, it would make more sense to call the result of fetching the tabel data 'tableData' ( as before) and using the store to manage the data. 

Exposing the low-level request-result to the caller leads to the issue where the caller does not property handle the result resulting in the null pointer that can be seen using the preview

Reply:
Refactored to add page to tableData